frontend/public/components/search.tsx

```diff
@@ -140,6 +140,7 @@ const SearchPage_: React.FC<SearchProps> = (props) => {
   const removeLabelFilter = (value: string) => {
     const newLabels = labelFilter.filter((keepItem: string) => keepItem !== value);
     setLabelFilter(newLabels);
+    setSelector(selectorFromString(newLabels.join(',')));
```

Comment:
Hey, @zherman0. I think the real problem here is that we're tracking both `labelFilter` and `selector` separately in state, which means they can get out of sync. I'd recommend only tracking `labelFilter` and then calling `selectorFromString` in the one spot we need when rendering. That way we're not tracking the same value twice and prevents bugs like this.

Reply:
Agreed.  I will update shortly.